server/lobbyconnection.py

```diff
@@ -895,66 +951,121 @@ def _set_user_agent_and_version(self, user_agent, version):
+                        table_mod.c.uid == uid
+                    ).limit(1)
+                )
 
                 row = await result.fetchone()
-                uid, name, version, author, ui, date, downloads, likes, played, description, filename, icon, likerList = (row[i] for i in range(13))
-                link = urllib.parse.urljoin(config.CONTENT_URL, "faf/vault/" + filename)
+
+                link = urllib.parse.urljoin(config.CONTENT_URL, "faf/vault/" + row["filename"])
+
                 thumbstr = ""
-                if icon:
-                    thumbstr = urllib.parse.urljoin(config.CONTENT_URL, "faf/vault/mods_thumbs/" + urllib.parse.quote(icon))
+                if row["icon"]:
+                    thumbstr = urllib.parse.urljoin(
+                        config.CONTENT_URL, "faf/vault/mods_thumbs/" 
+                        + urllib.parse.quote(row["icon"])
+                )
 
-                out = dict(command="modvault_info", thumbnail=thumbstr, link=link, bugreports=[],
-                           comments=[], description=description, played=played, likes=likes + 1,
-                           downloads=downloads, date=int(date.timestamp()), uid=uid, name=name, version=version, author=author,
-                           ui=ui)
+                out = dict(
+                    command="modvault_info", 
+                    thumbnail=thumbstr, 
+                    link=link, 
+                    bugreports=[],
+                    comments=[], 
+                    likes=row["likes"] + 1,
+                    date=int(row["date"].timestamp()),
+                )
+                out.update({
+                    key: row[key]
+                    for key in [
+                        "name", "version", "author", "ui", "downloads",
+                        "description", "played", "uid"
+                    ]
+                })
 
                 try:
-                    likers = json.loads(likerList)
+                    likers = json.loads(row["likers"])
                     if self.player.id in likers:
                         canLike = False
                     else:
                         likers.append(self.player.id)
-                except:
+                except Exception:
                     likers = []
 
                 # TODO: Avoid sending all the mod info in the world just because we liked it?
                 if canLike:
                     await conn.execute(
-                        "UPDATE mod_stats s "
-                        "JOIN mod_version v ON v.mod_id = s.mod_id "
-                        "SET s.likes = s.likes + 1, likers=%s WHERE v.uid = %s",
-                        json.dumps(likers), uid)
+                        mod_stats.update().values(
+                            likes=mod_stats.c.likes + 1,
+                            likers=json.dumps(likers)
+                        ).where(
+                            mod_version.c.uid == uid
+                        ).where(
```

Comment:
Imho you can write multiple and conditions in one where separated with comma

Reply:
https://docs.sqlalchemy.org/en/13/core/tutorial.html#multiple-table-updates

Reply:
That link is indeed what I was looking at.
Using `.where(and_(condition1, condition2))` also works, if you prefer it.